src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -458,6 +458,10 @@ bool os::supports_sse() {
 #endif // AMD64
 }
 
+bool os::supports_serialize(){
```

Comment:
This function is unnecessary and pollutes the OS namespace with something that is not OS related. Just use VM_Version::supports_serialize() directly.

Reply:
orderAccess_linux_x86.hpp is included in orderAccess.hpp, atomic.hpp, etc. VM_Version is not defined in any of the nested header files. If I add inclusion of vm_version.hpp in any of these nested head filer,  it will be messy.
Not sure about the best solution here.

Reply:
What actually happens if you just include vm_version.hpp in orderAccess_linux_x86.hpp ?

Reply:
it will give me errors, like  'Atomic' has not been declared.
Also, the cross_modify_fence_impl actually is os-dependent. Linux and Windows have different implementations.

Reply:
vm_version_x86.hpp has a dependency on universe.hpp which has a dependency on orderAccess.hpp which is why that circularity problem arises. That can be fixed by moving the inline definition of supports_clflush()  (which calls `Universe::is_fully_initialized()`) out of vm_version_x86.hpp into vm_version_x86.cpp. We can then proceed to included vm_version.hpp in orderAccess.hpp (something I'm surprised is not already done as I recall CPU specific memory barriers being used back in JDK 7 and 8). That change (of course) requires a few other tweaks, so I put the changes together here and ran them through our build system.
https://github.com/openjdk/jdk/compare/master...dholmes-ora:orderAccess-vm_version?expand=1

Reply:
cool. thanks.